core/mock_test.go

```diff
@@ -134,6 +137,12 @@ func (m mockASG) UpdateAutoScalingGroup(*autoscaling.UpdateAutoScalingGroupInput
 	return m.uasgo, m.uasgerr
 }
 
-func (m mockASG) DescribeTags(*autoscaling.DescribeTagsInput) (*autoscaling.DescribeTagsOutput, error) {
-	return m.dto, m.dterr
+func (m mockASG) DescribeTagsPages(input *autoscaling.DescribeTagsInput, function func(*autoscaling.DescribeTagsOutput, bool) bool) error {
+	function(m.dto, true)
+	return nil
+}
+
+func (m mockASG) DescribeAutoScalingGroupsPages(input *autoscaling.DescribeAutoScalingGroupsInput, function func(*autoscaling.DescribeAutoScalingGroupsOutput, bool) bool) error {
+	function(m.dasgo, true)
+	return nil
```

Comment:
It's for mocking out the call to the aws api, so instead of actually calling AWS's DescribeAutoScalingGroupsPages, what you are doing is calling the mock'd objects' implementation of the DescribeAutoScalingGroupsPages implementation, which passes in the canned output data

This page gives a better justice description that I can.

- https://aws.amazon.com/blogs/developer/mocking-out-then-aws-sdk-for-go-for-unit-testing/